lib/parse-client-options.js

```diff
@@ -66,7 +66,8 @@ function parseOptions (userOptions) {
   /* istanbul ignore else */
 
   if (!process.browser) {
-    clientDefaults.request.agent = getRequestAgent(defaults(userOptions, DEFAULTS))
+    userOptions.baseUrl = clientDefaults.baseUrl
+    clientDefaults.request.agent = getRequestAgent(userOptions)
```

Comment:
I would prefer

```js
clientDefaults.request.agent = getRequestAgent(clientDefaults.baseUrl , userOptions)
 ```

So we don’t mutate the options object passed in by the user. It requires a change in `get-request-agent.js` but this is not a public API so it’s no problem to change.

Thanks for your help!